app/models/hbx_enrollment.rb

```diff
@@ -1065,11 +1067,15 @@ def propagate_selection
   end
 
   def update_tax_household_enrollment
-    return if is_shop?
+    return if is_shop? || dental? || applied_aptc_amount.zero?
     return unless EnrollRegistry.feature_enabled?(:temporary_configuration_enable_multi_tax_household_feature)
 
-    TaxHouseholdEnrollment.where(enrollment_id: id).each do |th_enrollment|
-      th_enrollment.update_attributes(applied_aptc: (th_enrollment.available_max_aptc * elected_aptc_pct))
+    thh_enr_premiums = thh_enr_group_ehb_premium_of_aptc_members(aptc_tax_household_enrollments)
```

Comment:
Isn't this logic needs to be updated to take min of tax household ehb premium and tax household available aptc when enrollment ehb premium is less than total available aptc?

Reply:
This is handled in app/models/hbx_enrollment.rb#update_multiple_thh_enrollments

Reply:
we don't really need STHH vs MTHH differentiation here, logic we've in MTHH works for STHH as well. you can remove this.